classic-editor.php

```diff
@@ -434,17 +434,33 @@ public static function user_settings() {
 	}
 
 	public static function network_settings() {
-		$is_checked =  ( get_network_option( null, 'classic-editor-allow-sites' ) === 'allow' );
+		$editor = get_network_option( null, 'classic-editor-replace' );
+		$is_checked = ( get_network_option( null, 'classic-editor-allow-sites' ) === 'allow' );
 
 		?>
+		<h2><?php _e( 'Editor Settings', 'classic-editor' ); ?></h2>
 		<table class="form-table">
+			<?php wp_nonce_field( 'editor-settings', 'classic-editor-network-settings' ); ?>
 			<tr>
-				<th scope="row"><?php _ex( 'Classic Editor', 'Editor Name', 'classic-editor' ); ?></th>
+				<th scope="row"><?php _e( 'Default editor for all sites', 'classic-editor' ); ?></th>
 				<td>
-				<?php wp_nonce_field( 'allow-site-admin-settings', 'classic-editor-network-settings' ); ?>
-				<input type="checkbox" name="classic-editor-allow-sites" id="classic-editor-allow-sites" value="allow"<?php if ( $is_checked ) echo ' checked'; ?>>
-				<label for="classic-editor-allow-sites"><?php _e( 'Allow site admins to change settings', 'classic-editor' ); ?></label>
-				<p class="description"><?php _e( 'By default the Block Editor is replaced with the Classic Editor and users cannot switch editors.', 'classic-editor' ); ?></p>
+					<p>
+						<input type="radio" name="classic-editor-replace" id="classic-editor-classic" value="classic"<?php if ( $editor !== 'block' ) echo ' checked'; ?> />
+						<label for="classic-editor-classic"><?php _ex( 'Classic Editor', 'Editor Name', 'classic-editor' ); ?></label>
+					</p>
+					<p>
+						<input type="radio" name="classic-editor-replace" id="classic-editor-block" value="block"<?php if ( $editor === 'block' ) echo ' checked'; ?> />
+						<label for="classic-editor-block"><?php _ex( 'Block Editor', 'Editor Name', 'classic-editor' ); ?></label>
+					</p>
+					<p class="description error"><?php _e( 'Note: Setting the default network editor to Block Editor while disallowing site admins to change settings, defeats the purpose of these options.', 'classic-editor' ); ?></p>

```

Comment:
Hmmmm, not sure if we need this warning? Yes, the plugin has an option to "self-disable", it is there for completeness and for cases where it may be needed temporarily. Maybe can add it initially hidden, then show only when that combination is selected (with a bit of js)?

Reply:
IMHO is facoltative but not necessary: a warning and related translation that can generate confusion. :)

Reply:
I see your point. Maybe engage the self-disable option in such a case? Or would that be too intrusive?

Reply:
> Maybe can add it initially hidden, then show only when that combination is selected (with a bit of js)?
Considered that too. Or an admin message after saving...

> a warning and related translation that can generate confusion.
True. Maybe simplify the message? Or remove completely...

Or... what about simply making the combination impossible by forcing "Allow site admins..." to true if the network default is set to Block Editor ? :)